session.go

```diff
@@ -103,7 +103,7 @@ func addrsToHosts(addrs []string, defaultPort int) ([]*HostInfo, error) {
 }
 
 // NewSession wraps an existing Node.
-func NewSession(cfg ClusterConfig) (*Session, error) {
+func NewSession(cfg ClusterConfig) (_ *Session, err error) {
```

Comment:
I would just add `var err error` to L130 so no need for naming those.

https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

Reply:
I think it's more robust as it's easy to `return err2` somewhere later and make the leak pop up again

Reply:
I vote for adding var err error

Reply:
fine